mr/do_context.h

```diff
@@ -54,7 +96,8 @@ class RawContext {
  public:
   //! std/absl monostate is an empty class that gives variant optional semantics.
   using InputMetaData = absl::variant<absl::monostate, int64_t, std::string>;
-  using FreqMapRegistry = absl::flat_hash_map<std::string, std::unique_ptr<FrequencyMap<uint32_t>>>;
+  using FreqMapRegistry =
```

Comment:
Another way to do this is to whitelist possible types using variant of `std::unique_ptr<FrequencyMap<uint32_t>` and  `std::unique_ptr<FrequencyMap<string>` .